client/src/utils/uploadbox.js

```diff
@@ -245,13 +245,15 @@ export class UploadQueue {
                 }
             });
         }
+        // Returns last added file index.
+        return this.nextIndex - 1;
     }
 
     // Remove file from queue
-    remove(index) {
+    remove(index, fileSetKey) {
         // Removes the item identified by index.
```

Comment:
As mentioned above, we should grab the file from the queue here, generate the fileSetKey and remove as you correctly do.